posthog/gzip_middleware.py

```diff
@@ -0,0 +1,35 @@
+import re
+from typing import List
+
+from django.conf import settings
+from django.middleware.gzip import GZipMiddleware
+
+
+class InvalidGzipAllowList(Exception):
+    pass
+
+
+def allowed_path(path: str, allowed_paths: List) -> bool:
+    return any(pattern.search(path) for pattern in allowed_paths)
+
+
+class PostHogGZipMiddleware(GZipMiddleware):
```

Comment:
Nit: it doesn't seem super informational to use $PRODUCT_NAME as the prefix for a subclass, as it's pretty obvious that this is a PostHog-specific. In this case maybe something like `ScopedGZipMiddleware` could be more meaningful?